wallet/src/main/java/bisq/wallet/GrpcWalletService.java

```diff
@@ -0,0 +1,248 @@
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class GrpcWalletService implements WalletService {
+
+    public record ServerConfig(String host, int port) {
+    }
+
+    private final ManagedChannel channel;
+    private final WalletServiceGrpc.WalletServiceBlockingStub blockingStub;
+    private final WalletServiceGrpc.WalletServiceFutureStub futureStub;
+
+    private final Observable<Coin> balance = new Observable<>();
+    private final ObservableSet<Transaction> transactions = new ObservableSet<>();
+    private final ObservableSet<String> walletAddresses = new ObservableSet<>();
+
+    public GrpcWalletService(ServerConfig config) {
+        this.channel = ManagedChannelBuilder.forAddress(config.host(), config.port())
+                .usePlaintext()
+                .build();
+        this.blockingStub = WalletServiceGrpc.newBlockingStub(channel);
+        this.futureStub = WalletServiceGrpc.newFutureStub(channel);
+    }
+
+    @Override
+    public void encryptWallet(String password) {
+        EncryptWalletRequest request = EncryptWalletRequest.newBuilder().setPassword(password).build();
+        blockingStub.encryptWallet(request);
+    }
+
+    @Override
+    public CompletableFuture<List<String>> getSeedWords() {
+        GetSeedWordsRequest request = GetSeedWordsRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.getSeedWords(request))
+                .thenApply(GetSeedWordsResponse::getSeedWordsList);
+    }
+
+    @Override
+    public boolean isWalletReady() {
+        IsWalletReadyRequest request = IsWalletReadyRequest.newBuilder().build();
+        return blockingStub.isWalletReady(request).getReady();
+    }
+
+    @Override
+    public CompletableFuture<String> getUnusedAddress() {
+        GetUnusedAddressRequest request = GetUnusedAddressRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.getUnusedAddress(request))
+                .thenApply(GetUnusedAddressResponse::getAddress);
+    }
+
+    @Override
+    public ObservableSet<String> getWalletAddresses() {
+        return walletAddresses;
+    }
+
+    @Override
+    public CompletableFuture<ObservableSet<String>> requestWalletAddresses() {
+        GetWalletAddressesRequest request = GetWalletAddressesRequest.newBuilder().build();
+        return toCompletableFuture(futureStub.getWalletAddresses(request))
+                .thenApply(response -> {
+                    walletAddresses.clear();
+                    walletAddresses.addAll(response.getAddressesList());
+                    return walletAddresses;
+                });
+    }
+
+    @Override
+    public CompletableFuture<List<? extends TransactionInfo>> listTransactions() {
```

Comment:
I dont think we will have multiple impl for TransactionInfo -> make a class out of TransactionInfo. Was leftover from multi wallets support.